opm/common/utility/TimeService.hpp

```diff
@@ -32,6 +32,13 @@ namespace Opm {
             int year{0};
             int month{0};
             int day{0};
+
+            bool operator==(const YMD& data) const
+            {
+                return year == data.year &&
+                       month == data.month &&
+                       day == data.day;
+            }
```

Comment:
Is there a reason the compiler-generated comparison doesn't work?

Reply:
>Is there a reason the compiler-generated comparison doesn't work?

Is there a compiler generated comparison?

Reply:
> > Is there a reason the compiler-generated comparison doesn't work?
> 
> Is there a compiler generated comparison?

There is not.  I got confused with the assignment operator :man_facepalming:

C++20 will help a little bit here as we'll be able to write
```C++
struct YMD {
    int year, month, day; 

    auto operator<=>(const YMD&) const = default;
};
```
Of course, in C++20 there's also the **much** improved [`year_month_day`](https://en.cppreference.com/w/cpp/chrono/year_month_day) library type so we won't have to use a home-grown implementation.